### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by RickyAvina**
`query()` successfully find free periods for all mandatory members if such times exist. Currently passing all given and custom tests. Optional attendees not yet accounted for.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java [hunk 1]
```diff
@@ -149,6 +156,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/home");
   }
 
+  /**
+   * Check whether a given string contains javascript or dangerous code.
+   * @param str - a string to validate
+   * @return - bool that represents whether `str` contains dangerous characters
```
Comment (cassiehan) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java:162]: @return true if the input `str` contains dangerous characters

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java [hunk 2]
```diff
@@ -149,6 +156,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+   */
+  private static boolean checkValidate(String str) {
+        if (str != null) {
+            String patternString = ".*javascript:.*|.*//.*|.*<.*|.*>.*";
```
Comment (cassiehan) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java:166]: make it a constant

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,136 @@
...
+    Collection<TimeRange> freePeriods = new ArrayList<TimeRange>();
+    while (startTime < TimeRange.END_OF_DAY) {
+        int endTime = getEndTime(startTime, timeRanges); // find upper limit of free time
+        // add to free times
+        TimeRange proposedTime = TimeRange.fromStartEnd(startTime, endTime, true);
+        freePeriods.add(proposedTime);
+        System.out.println("Proposed time: " + proposedTime);
+        
+        if (endTime >= TimeRange.END_OF_DAY)
+            break;
```
Comment (cassiehan) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:61]: This structure is quite wired. You could make it either 
if (endTime >= TimeRange.END_OF_DAY) {
  break;
}
or
if (endTime >= TimeRange.END_OF_DAY) break;

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 4]
```diff
@@ -147,6 +148,34 @@ public void overlappingEvents() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void twoOverlaps() {
```
Comment (cassiehan) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:152]: It is recommended to make a test in the following structure:
preparation

function call

assert

In this case:
Collection<Event> events = Arrays.asList(...);
MeetingRequest request = ...;

Collection<TimeRange> actual = query.query(events, request);

Collection<TimeRange> expected =...
Assert.assertEquals(expected, actual);

Comment (RickyAvina) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:152]: Are you proposing a new test, or proposing I should restructure this test? I think this particular test follows the structure you mention here.

**Pull request merged**